### ISSUE TITLE
pytorch 1.3, addition of different data typed tensors after permute giving incorrect results

### ISSUE THREAD

**Issue by vadapalliravikumar** [pytorch/pytorch#28747]
## 🐛 Bug

- Addition of different data typed tensors after permute giving incorrect results.
- Addition of same data typed tensors is fine
- Addition of different datatyped tensors was not allowed in pytorch 1.2, so this never happened in 1.2 version.

## To Reproduce

```>>> 
>>> t = torch.tensor([[1, 2], [5, 6]])
>>> 
>>> t1 = t.permute(1,0)
>>> 
>>> m = torch.tensor([[1.0], [2.0]])
>>> 
>>> 
>>> t
tensor([[1, 2],
        [5, 6]])
>>> t.dtype
torch.int64
>>>
>>> t1
tensor([[1, 5],
        [2, 6]])
>>> t1.dtype
torch.int64
>>>
>>> m
tensor([[1.],
        [2.]])
>>> m.dtype
torch.float32
>>> 
>>> t+m
tensor([[2., 3.],
        [7., 8.]])
>>> 
>>> t1+m
tensor([[2., 3.],
        [7., 8.]])
>>> 
```

## Expected behavior

```
>>>t1+m
 tensor([[2., 6.],
         [4., 8.]]
 ```

## Environment
PyTorch version: 1.3.0
Is debug build: No
CUDA used to build PyTorch: 10.1.243

OS: Ubuntu 16.04.6 LTS
GCC version: (Ubuntu 5.4.0-6ubuntu1~16.04.11) 5.4.0 20160609
CMake version: Could not collect

Python version: 3.6
Is CUDA available: Yes
CUDA runtime version: 10.1.243
GPU models and configuration: GPU 0: GeForce GTX 1080 Ti
Nvidia driver version: 418.87.01
cuDNN version: /usr/lib/x86_64-linux-gnu/libcudnn.so.7.4.1

Versions of relevant libraries:
[pip3] numpy==1.17.3
[pip3] torch==1.3.0
[pip3] torchvision==0.4.1
[conda] Could not collect

## Additional context

workaround:
```
>>>t1 = t.permute(1,0).contiguous()

>>> t1
tensor([[1, 5],
        [2, 6]])
>>> m
tensor([[1.],
        [2.]])
>>> t1+m
tensor([[2., 6.],
        [4., 8.]])
>>> 
```

cc @ezyang @gchanan @zou3519

**Comment by fmassa** [comment]
Seems like a pretty serious bug.

**Comment by koelscha** [comment]
This is not only happening for addition, but also for other operands, cf. reference above

**Comment by zou3519** [comment]
cc @nairbv

**Comment by gchanan** [comment]
This looks to be fixed on master.

```
>>> t1+m
tensor([[2., 6.],
        [4., 8.]])
```

**Comment by vadapalliravikumar** [comment]
@gchanan , what is the best way to consume the fix ? I am on the 1.3 release.

- Is it a simple fix that can be manually patched on top of 1.3 installation ? If yes, can u pls give a pointer to the fix.
- Otherwise, should I go for a full private installation from master ? And is master stable enough ?

**Comment by nairbv** [comment]
FYI this is the PR that fixes this issue: https://github.com/pytorch/pytorch/pull/28253

It impacts mixed-type (promoting) operations on non-contiguous tensors (so converting to same dtype or calling `.contiguous()` can workaround the issue).

**Comment by vadapalliravikumar** [comment]
thanks @nairbv 

this is such a critical issue. It may not affect those upgrading from 1.2 to 1.3. But it has a good chance of affecting those playing with named tensors and mixed typing in 1.3, please consider doing a patch release (1.3.0.x).

**Comment by soumith** [comment]
we're planning to do a patch release (1.3.1) because of this. Can do as early as this week, or early next week. finalizing details.

**Comment by vadapalliravikumar** [comment]
awesome! thank you @soumith

**Comment by ddkang** [comment]
Why was this issue closed? I don't believe 1.3.1 has been released?

**Comment by nairbv** [comment]
> Why was this issue closed? I don't believe 1.3.1 has been released?

Fixed on master. See here for 1.3.1 release tracking:
https://github.com/pytorch/pytorch/issues/28919
